app/uk/gov/hmrc/essttpstubs/services/TtpEligibilityActor.scala

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.essttpstubs.services
+
+import akka.actor.typed.scaladsl.Behaviors
+import akka.actor.typed.{ActorRef, Behavior}
+import uk.gov.hmrc.essttpstubs.model.{OverduePayments, TaxID}
+import uk.gov.hmrc.essttpstubs.services.EligibilityService.{EligibilityError, ServiceError}
+
+object TtpEligibilityActor {
+
+  sealed trait Command
+
+  case class MapPayeError(taxId: TaxID, error: EligibilityError, replyTo: ActorRef[Either[ServiceError, Unit]]) extends Command
+
+  case class MapFinancialData(taxId: TaxID, data: OverduePayments, replyTo: ActorRef[Either[ServiceError, Unit]]) extends Command
+
+  case class FindEligibilityData(taxId: TaxID, replyTo: ActorRef[Either[ServiceError, OverduePayments]]) extends Command
+
+  def apply(): Behavior[Command] = Behaviors.setup[Command] { ctx =>
+    ctx.log.info("starting the ttp eligibility actor")
```

Comment:
Wouldn't be easier to use `AtomicRef[TTP]` and simple function in service? The performance will be the same. The code will be much easier to understand.